helm/frontend/templates/deployment.yaml

```diff
@@ -0,0 +1,73 @@
+          image: "{{ .Values.deployment.image.repository }}:{{ .Values.deployment.image.tag }}"
+          ports:
+            - containerPort: {{ .Values.deployment.containerPort }}
+          env:
+            - name: ASSET_HOST
+              value: "{{ .Values.deployment.appAssetHost | default (printf "https://www-origin.%s.%s.%s" .Release.Namespace .Values.govukEnvironment .Values.govukDomainExternal) }}"
+            - name: GOVUK_APP_DOMAIN
+              value: "{{ .Values.deployment.appGovukAppDomain | default (printf "%s.%s.%s" .Release.Namespace .Values.govukEnvironment .Values.clusterDomain) }}"
+            - name: GOVUK_APP_DOMAIN_EXTERNAL
+              value: "{{ .Values.deployment.appGovukDomainExternal | default (printf "%s.%s.%s" .Release.Namespace .Values.govukEnvironment .Values.govukDomainExternal) }}"
+            - name: GOVUK_APP_NAME
+              value: "{{ .Values.deployment.appGovukAppName | default .Release.Name }}"
+            - name: GOVUK_APP_TYPE
+              value: "{{ .Values.deployment.appGovukAppType }}"
+            - name: GOVUK_WEBSITE_ROOT
+              value: "{{ .Values.deployment.appGovukWebsiteRoot | default (printf "https://www-origin.%s.%s.%s" .Release.Namespace .Values.govukEnvironment .Values.govukDomainExternal) }}"
+            - name: DEFAULT_TTL
+              value: "{{ .Values.deployment.appDefaultTtl }}"
+            - name: GOVUK_ASSET_ROOT
+              value: "{{ .Values.deployment.appGovukAssetRoot | default (printf "https://assets.%s.%s.%s" .Release.Namespace .Values.govukEnvironment .Values.govukDomainExternal) }}"
+            - name: GOVUK_CONTENT_SCHEMAS_PATH
+              value: "{{ .Values.deployment.appGovukContentSchemasPath }}"
+            - name: GOVUK_ENVIRONMENT
+              value: "{{ .Values.govukEnvironment }}"
+            - name: PLEK_SERVICE_CONTENT_STORE_URI
+              value: "{{ .Values.deployment.appPlekServiceContentStoreUri | default (printf "http://content-store.%s.%s" .Release.Namespace .Values.clusterDomain) }}"
+            - name: PLEK_SERVICE_PUBLISHING_API_URI
+              value: "{{ .Values.deployment.appPlekServicePublishingApiUri | default (printf "http://publishing-api-web.%s.%s" .Release.Namespace .Values.clusterDomain) }}"
+            - name: PLEK_SERVICE_SIGNON_URI
+              value: "{{ .Values.deployment.appPlekServiceSignonUri | default (printf "http://signon.%s.%s" .Release.Namespace .Values.clusterDomain) }}"
+            - name: PLEK_SERVICE_STATIC_URI
+              value: "{{ .Values.deployment.appPlekServiceStaticUri | default (printf "http://static.%s.%s" .Release.Namespace .Values.clusterDomain) }}"
+            - name: PORT
+              value: "{{ .Values.deployment.containerPort }}"
+            - name: RAILS_ENV
+              value: "{{ .Values.deployment.appRailsEnv }}"
+            - name: SECRET_KEY_BASE
+              value: "secret" #TODO: fix when we know how to handle secrets
```

Comment:
Just FYI, more on Secrets coming soon, but this will almost certainly look like:

```
- name: SECRET_KEY_BASE
   valueFrom:
       secretKeyRef:
           name: frontend-secrets
           key: SECRET_KEY_BASE
```

with the Kubernetes secret created/managed by Secrets Store backed by AWS Secrets Manager.

([secrets and env var docs here](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables))

Reply:
cool

Reply:
Actually, after a brief chat with @sengi just now, it's probably best to define this secret as a `Secret` resource just now - our approach to handling secrets overall should end with a `Secret` object being populated, so having the Deployment refer to a Secret just now should mean fewer changes down the line.